lib/elastic.js

```diff
@@ -116,10 +112,55 @@ function _get_mapping(client, index, type) {
     }
 }
 
+function create_index_if_not_exists_with_type(client, index, type) {
+    if (ALREADY_CREATED_INDICES[index]) { return Promise.resolve(); }
+
+    return _check_index_exists(client, index)
+        .then((exists) => {
+            if (!exists) {
+                return _create_index(client, index, type);
+            }
+        })
+        .then(() => {
+            ALREADY_CREATED_INDICES[index] = true;
+        });
+}
+
+function _check_index_exists(client, index) {
+    var options = {index: index};
+    if (client instanceof AmazonElasticsearchClient) {
+        return client.indexExistsAsync(options);
+    } else {
+        return client.indices.exists(options);
+    }
+}
+
+function _create_index(client, index, type) {
+    var options = {
+        index: index,
+        body: {
+            mappings: {}
+        }
+    };
+    options.body.mappings[type] = DYNAMIC_MAPPING_SETTINGS;
+    if (client instanceof AmazonElasticsearchClient) {
+        return client.createIndexAsync(options);
```

Comment:
Just curious: instead of `if (client instanceof AmazonElasticsearchClient)` in a bunch of places, does it make sense for a client super class to figure this out and choose the correct function to call?